### ISSUE TITLE
Dark mode doesn't work in Chromium-based browsers

### ISSUE THREAD

**Issue by lah7** [ubuntu-mate/ubuntu-mate.org#63]
Chromium should support the [prefers-color-scheme CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/), but it doesn't appear to be working on Linux browsers when the desktop is using a dark theme.

**Comment by Dorson** [comment]
The latest Brave Browser does that.  Same of better web experience.

**Comment by Dorson** [comment]
It's also very nice to allow the users to set their own colors.

Recently, I coded the Dark Mode switch in CSS and basic Java Script.

In my version of the dark mode, we can have multiple color themes in one CSS file. Then set or remember color settings with one short JS function without logins.

**Try it** on the GitHub demo page :

https://dorson.github.io/CSS-Dark-Mode-and-color-switch/

Feel free to fork, copy, adopt, etc..

Have fun ! :+1:

**Comment by lah7** [comment]
Since there is no obvious "fix" for these browsers, I'll add a manual toggle in the footer. The toggle will only appear if the browser doesn't support themes automatically, and JavaScript is on.

If the browser supports this later, then the dark mode will be automatic.

[I'm aware of CSS variables](https://github.com/ubuntu-mate/start.ubuntu-mate.org/commit/8e83fe50eb117b146ade45145a13c777ffb4f5b2), so thanks for the reminder. Setting colours would be nice for applications, but doesn't quite fit with the scope of this website. To avoid re-engineering SCSS variables and try to be compatible with pre-2016 browsers, I went with a dual style sheets approach for this website.

In short, it's working for Chrome on Windows 10, but not Chrome on Ubuntu MATE (with Ambiant-MATE-Dark). Firefox works perfectly.

**Comment by Dorson** [comment]
> I went with a dual style sheets approach for this website.

Also good. Multiple CSS files as a solution does work well for ages. The upcoming **@media CSS rules** for exception cases will make it less important as it was in the past, or now.

> colors in CSS variables

In my opinion they are very nice and handy in the long term. Imagine returning to edit that CSS file years later on.

The set of simple color variables is faster to change later on, if it's all in one place, at the start of the CSS file.

like:

```
:root {
--color-background : grey ; 
--color-background-contrast  : coral ; 
--color-background-hi  : green ; 
--color-background-low  : olive ; 
--color-text  : black ; 
--color-link  : blue ; 
--color-brighter  : white ; 
--color-colorful  : pink ; 
--color-click-items  : red ; 
}
```

CSS color variables are nice in the long term, because we can also change them in defined exception cases later on. That is where the **@media CSS rules** will come into place, instead of multiple CSS files.

The CSS variables are the first step into the future, where the CSS rules can become more dynamic, and less dependable on server processing or JS functions, or events.

 > the CSS media rule for : prefers-color-scheme : dark 

It's almost at the breaking point of being widely adopted. Works on the the latest mobile browsers, which is already at 40%-60% of all casual website visitors.

I kind of like this CSS rule for simplicity of it.

The light intensity one will be even better for mobile devices. As of now, not so much supported, but it will be.

```
@media screen and (light-level: washed) {
  p {
    background: white;
    color: black;
    font-size: 2em;
  }
}
```


Mozilla has a great list of the upcoming CSS media rules:

https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features
